### PR TITLE
gyp: update gyp to 0.2.0

### PR DESCRIPTION
**Pull request by ryzokuken**
##### Checklist

- [ ] `npm install && npm test` passes
- [ ] tests are included 
- [ ] documentation is changed or added
- [ ] commit message follows [commit guidelines](https://github.com/nodejs/node/blob/master/doc/guides/contributing/pull-requests.md#commit-message-guidelines)

##### Description of change

/cc @targos @cclauss @nodejs/gyp @nodejs/node-gyp 

For some reason, I don't have write access to this repo? Is it not open to collaborators? :sweat_smile:

### CODE REVIEW

**Comment by ryzokuken** [comment]
All tests pass locally! :tada:

**Comment by rvagg** [comment]
This is gyp-next, right? I suppose we should also update the README to document the change and point to the new upstream in this PR.

This isn't open to nodejs/node collaborators, it's a separate project with a separate team

File: gyp/.flake8 [hunk 1]
```diff
@@ -0,0 +1,4 @@
+[flake8]
```
Comment (cclauss) [inline gyp/.flake8:1]: I am not a fan of these files because the hide configuration details and remove the flexibility to add different tests in different locations in the workflow.  Explicit is better that implicit so put this all in the flake8 command.

I don’t think we want to give contributors an open license to add even more complex code than we already have.  It would be more prudent to increase max-complexity to our current level and remove the C901.

File: gyp/.github/workflows/Python_tests.yml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  Python_tests:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      max-parallel: 15
+      matrix:
+        os: [macos-latest, ubuntu-latest] # , windows-latest]
+        python-version: [2.7, 3.6, 3.7, 3.8] # 3.5,
+    steps:
+      - uses: actions/checkout@v1
```
Comment (cclauss) [inline gyp/.github/workflows/Python_tests.yml:17]: v2 ?

File: gyp/.github/workflows/Python_tests.yml [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v1
+        with:
+          python-version: ${{ matrix.python-version }}
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install -r requirements_dev.txt
+      - name: Lint with flake8
+        run: flake8 . --count --show-source --statistics
```
Comment (cclauss) [inline gyp/.github/workflows/Python_tests.yml:27]: Be explicit by putting ALL options here.  Having some options here and others in a dot file bugs me.

File: gyp/.gitignore [hunk 4]
```diff
@@ -1 +1,143 @@
-*.pyc
```
Comment (cclauss) [inline gyp/.gitignore:1]: Please add those pesky Python `.DS_Store` files.

File: gyp/PRESUBMIT.py [hunk 5]
```diff
@@ -1,138 +0,0 @@
-# Copyright (c) 2012 Google Inc. All rights reserved.
```
Comment (cclauss) [inline gyp/PRESUBMIT.py:1]: It would be good to have a https://pre-commit.com but we can do that in a separate PR.

File: gyp/LICENSE [hunk 6]
```diff
@@ -1,3 +1,4 @@
+Copyright (c) 2019 Ujjwal Sharma. All rights reserved.
```
Comment (cclauss) [inline gyp/LICENSE:1]: Question: Should we add Node.js Contributors?  I am happy either way.

**Comment by cclauss** [comment]
Travis CI fails.  Are we done with Travis?

File: gyp/LICENSE (hunk 6 above)
Comment (rvagg) [reply gyp/LICENSE:1]: Yeah, that's a bit weird, and it's in nodejs/gyp-next too. Can we get that fixed so it doesn't look like a personal project?

**Comment by ryzokuken** [comment]
> This is gyp-next, right? I suppose we should also update the README to document the change and point to the new upstream in this PR.

@rvagg sounds good, will do.

@cclauss all your comments are very valid. However, I don't believe this is the right place to discuss or address them. This is a simple PR upgrading the gyp version in the codebase from the old one to the v0.2.0 release of gyp-next. I'd recommend that you voice all your concerns gyp-next repo and we'd love to address them all and make sure they're included in the next release.

**Comment by cclauss** [comment]
How is node-gyp@0.2.0 related to https://github.com/nodejs/node-gyp/releases?  I am a bit confused.

**Comment by richardlau** [comment]
> 
> 
> How is node-gyp@v0.2.0 related to https://github.com/nodejs/node-gyp/releases?
> I am a bit confused.

@cclauss it isn't. This PR updates the embedded Python based gyp tool from https://github.com/nodejs/gyp-next into this project (node-gyp) which is a Node.js wrapper around gyp to build Node.js addons. The 0.2.0 refers to the version of gyp-next and not the version of node-gyp.

**Comment by ryzokuken** [comment]
I have pushed a commit that addresses @rvagg's comment about the documentation. @cclauss do you agree that we should address your concerns on the `gyp-next` issue tracker since they're off-topic for this thread?

**Comment by rvagg** [comment]
I'm still concerned about this only carrying your personal copyright notice rather than the org, or at least whoever else is/does contribute. I'd rather it be org, like nodejs/node. Or is there a good reason for this?

File: README.md [hunk 7]
```diff
@@ -4,9 +4,9 @@
 [![Build Status](https://github.com/nodejs/node-gyp/workflows/Python_tests/badge.svg)](https://github.com/nodejs/node-gyp/actions?workflow=Python_tests)
 
 `node-gyp` is a cross-platform command-line tool written in Node.js for
-compiling native addon modules for Node.js. It contains a fork of the
-[gyp](https://gyp.gsrc.io) project that was previously used by the Chromium
-team, extended to support the development of Node.js native addons.
+compiling native addon modules for Node.js. It contains a checkout of the
```
Comment (rvagg) [inline README.md:7]: I think "vendored copy of the..." might be more idiomatic for what we're doing

**Comment by rvagg** [comment]
can we get an updated version of this and the minor issues pointed out addressed? then we can bundle in a v7 and just get it done.

**Comment by ryzokuken** [comment]
@rvagg sure. Sorry I was a little busy this last week. I'm on it now though.

**Comment by ryzokuken** [comment]
@rvagg sorry for the delay, take a look and let me know if all's good and I'll autosquash the commits ready to be merged. :smile:

File: gyp/.flake8 (hunk 1 above)
Comment (rvagg) [reply gyp/.flake8:1]: @cclauss I think this critique should be taken over to gyp-next since we're just vendoring in that code, no?

File: gyp/.github/workflows/Python_tests.yml (hunk 2 above)
Comment (rvagg) [reply gyp/.github/workflows/Python_tests.yml:17]: ditto, a critique for nodejs/gyp-next

File: gyp/.gitignore (hunk 4 above)
Comment (rvagg) [reply gyp/.gitignore:1]: ditto, gyp-next, shouldn't impact us I think

**Comment by rvagg** [comment]
* a6b76a8 gyp: update gyp to 0.2.1
* e529f33 doc: update README to reflect upgrade to gyp-next
* ebc34ec gyp: update gyp to 0.2.0
